packages/ra-data-graphql-simple/src/buildGqlQuery.ts

```diff
@@ -241,27 +240,5 @@ export const buildApolloArgs = (
     return args;
 };
 
-export const getArgType = (arg: IntrospectionInputValue): TypeNode => {
-    const type = getFinalType(arg.type);
-    const required = isRequired(arg.type);
-    const list = isList(arg.type);
-
-    if (list) {
-        if (required) {
-            return gqlTypes.listType(
-                gqlTypes.nonNullType(
-                    gqlTypes.namedType(gqlTypes.name(type.name))
-                )
-            );
-        }
-        return gqlTypes.listType(gqlTypes.namedType(gqlTypes.name(type.name)));
-    }
-
-    if (required) {
-        return gqlTypes.nonNullType(
-            gqlTypes.namedType(gqlTypes.name(type.name))
-        );
-    }
-
-    return gqlTypes.namedType(gqlTypes.name(type.name));
-};
+export const getArgType = (arg: IntrospectionInputValue): TypeNode =>
+    getGqlType(arg.type);
```

Comment:
Why do you extract the content of the function in another one? Now getArgType seems to be an empty shell. You should move back the content of etGqlType there.

Reply:
It's because `getGqlType` is a recursive function. `getArgType` has a different parameter, `IntrospectionInputValue`, while the recursive one is called with the type, `IntrospectionType | IntrospectionListTypeRef | IntrospectionTypeRef`.

Although after checking again I see `getArgType` is only called in one place. Should I move the call inline above? https://github.com/marmelab/react-admin/blob/fccba79c118082be6d8c40ba5322ff82f02aaa56/packages/ra-data-graphql-simple/src/buildGqlQuery.ts#L235
So it would be:

```ts
gqlTypes.variableDefinition(
    gqlTypes.variable(gqlTypes.name(arg.name)),
    getGqlType(arg.type)
),
```